src/components/App.tsx

```diff
@@ -796,30 +804,37 @@ export class App extends React.Component<any, AppState> {
                     // if shift is not clicked, this will always return true
                     // otherwise, it will trigger selection based on current
                     // state of the box
-                    if (!hitElement.isSelected) {
-                      hitElement.isSelected = true;
+                    if (!this.state.selectedElementIds[hitElement.id]) {
+                      this.setState(prevState => ({
+                        selectedElementIds: {
+                          ...prevState.selectedElementIds,
+                          [hitElement!.id]: true,
+                        },
+                      }));
                       elements = elements.slice();
                       elementIsAddedToSelection = true;
                     }
 
                     // We duplicate the selected element if alt is pressed on pointer down
                     if (event.altKey) {
-                      elements = [
-                        ...elements.map(element => ({
-                          ...element,
-                          isSelected: false,
-                        })),
-                        ...getSelectedElements(elements).map(element => {
-                          const newElement = duplicateElement(element);
-                          newElement.isSelected = true;
-                          return newElement;
-                        }),
-                      ];
+                      // Move the currently selected elements to the top of the z index stack, and
+                      // put the duplicates where the selected elements used to be.
```

Comment:
I don't believe this is the expected behavior. When you alt drag, you want to create a fresh copy of the element and move them. With your change, the initial elements slightly change shape when doing this operation.

I don't think it's the end of the world but I would like to preserve the previous behavior where the elements that are being moved are the one that slightly change shape.